tests/test_app.py

```diff
@@ -222,10 +338,46 @@ def test_check_shutdown():
 
 
 def test_check_bad_shutdown():
-    print([sys.executable, os.path.join(os.path.dirname(__file__), "miniapp.py"), "bad"])
-    process = do_run([sys.executable, os.path.join(os.path.dirname(__file__), "miniapp.py"), "bad"])
+    print(
+        [sys.executable, os.path.join(os.path.dirname(__file__), "miniapp.py"), "bad"]
+    )
+    process = do_run(
+        [sys.executable, os.path.join(os.path.dirname(__file__), "miniapp.py"), "bad"]
+    )
     out, err = do_kill(process, killtime=5, termtime=2)
     print(out, err)
     assert "----- Thread Dump ----" in out
     assert "STOP" not in out
     assert "SHUTDOWN COMPLETE" not in out
+
+
+def test_compiler_exception_output(snippetcompiler):
+    snippetcompiler.setup_for_snippet_external(
+        """
+entity Test:
+    number attr
+end
+
+implement Test using std::none
+
+o = Test(attr="1234")
+"""
+    )
+
+    output = """Could not set attribute `attr` on instance `__config__::Test (instantiated at ./main.cf:8)` """ \
+        """(reported in Construct(Test) (./main.cf:8))
+caused by:
+  Invalid value '1234', expected Number (reported in Construct(Test) (./main.cf:8))
+"""
+
+    def exec(*cmd):
+        process = do_run(
+            [sys.executable, "-m", "inmanta.app"] + list(cmd),
+            cwd=snippetcompiler.project_dir,
+        )
+        out, err = process.communicate(timeout=5)
+        assert out.decode() == ""
+        assert err.decode() == output
```

Comment:
Wouldn't it be better to check whether the outputstream contains output, compared to the check for an exact match?

Reply:
the bug is about getting the same on export and compile, so I either check exact or check similarity,...

Reply:
Ok. I see. I was just concerned that we might break the test case when the export or the compile command produces more output in the future.